Base/Python/slicer/util.py

```diff
@@ -780,13 +780,26 @@ def loadColorTable(filename, returnNode=False):
 
 
 def loadFiberBundle(filename, returnNode=False):
-    """Load node from file.
+    """Load fiber bundle node from file.
+
+    .. warning::
+    To ensure the FiberBundleFile reader is registered, the ``SlicerDMRI``
+    extension needs to be installed.
 
     :param filename: full path of the file to load.
     :param returnNode: Deprecated.
     :return: loaded node (if multiple nodes are loaded then a list of nodes).
       If returnNode is True then a status flag and loaded node are returned.
     """
+
+    if slicer.app.ioManager().registeredFileReaderCount("FiberBundleFile") == 0:
+        errorMessage = f"FiberBundleFile reader not registered: failed to read data from file: {filename}"
+        errorMessage += "\n" + "Try to install the SlicerDMRI extension"
```

Comment:
“Try to install…” seems to invoke a lack of confidence that it might not work. “Install the SlicerDMRI extension…” would be more confident.

Further, checking if the SlicerDMRI is already installed would be helpful. Checking if the file reader is present is a transitive check but not a direct check to see if SlicerDMRI is already installed.

You could also prompt the user directly install the extension as some may not know how to install an extension if they are told this information. https://slicer.readthedocs.io/en/latest/developer_guide/script_repository.html#download-and-install-extension provides an example of such.

Reply:
Re recent push force:
- Added a check for the extension.
- Made the first message be imperative.
- The second one is less sure to be fixed by reinstalling the extension, so left the suggestion mode.
- I am hesitant to trigger an extension installation, especially if this is done silently (not sure if the command involves a prompt) and if this involves restarting the software. If the user does not know how to install an extension, I do not think a loading method should solve that, as it is an essential feature of 3D Slicer that users should be aware of by other means. But let me know if it should still be that way (should probably be made into a policy for any file type then).